pilot/pkg/model/push_context.go

```diff
@@ -695,6 +695,21 @@ func (ps *PushContext) ServiceForHostname(proxy *Proxy, hostname host.Name) *Ser
 	return nil
 }
 
+// ServicesForHostname returns a list of services that fall under the hostname provided. This hostname
+// can be a wildcard.
+func (ps *PushContext) ServicesForHostname(proxy *Proxy, hostname host.Name) []*Service {
+	if proxy == nil || proxy.SidecarScope == nil {
+		return nil
+	}
+	services := make([]*Service, 0)
```

Comment:
should we cache this in `sidecarscope` like `matchingServicesByHostName` and use it here instead of recomputing every time?

Reply:
will do in follow-up